app/src/main/java/com/example/coopertest/Results.kt

```diff
@@ -0,0 +1,163 @@
+package com.example.coopertest
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.provider.Settings.Global.getString
+import androidx.preference.PreferenceManager
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+
+class Results (meters : Double, context: Context){
+
+    var meters : Double=0.0
+    var age: Int=0
+    var athlete: Boolean = false
+    var gender : String = ""
+
+    init {
+        val mSharedPreference = PreferenceManager.getDefaultSharedPreferences(context)
+        val dateBirthday = mSharedPreference.getString("birthday", "null")
+        age=getAge(dateBirthday.toString())
+        this.meters=meters
+        athlete = mSharedPreference.getBoolean("athlete", false)
+        gender = mSharedPreference.getString("gender", "Male")!!
+    }
+
+    private fun getAge(dobString: String): Int {
+
+        var date: Date? = null
+        val sdf = SimpleDateFormat("yyyy-MM-dd")
+        try {
+            date = sdf.parse(dobString)
+        } catch (e: ParseException) {
+            e.printStackTrace()
+        }
+
+        if (date == null) return 0
```

Comment:
Could this go into the catch block? It would save some space and maybe we wouldn't need to have the Date? type and asign null first

Reply:
I don't know how do this. But there is no reason to have this problem : I want to add a control as location before launching test, it will check all parameters.